In [2]:
# Importando bibliotecas
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import os


In [3]:
# Caminho dos Arquivos
path = 'dataset/'

# Carreguei todos os aquivos um um dicionario de df
arquivos = [a for a in os.listdir(path) if a.endswith('.csv')]
dataframes = {arq.replace('.csv', ''): pd.read_csv(os.path.join(path, arq)) for arq in arquivos}

### O que fiz acima foi:
 1. Definir uma variável com o caminho para os arquivos csvs
 2. Criei a variável arquivos onde utilizei list comprehension para armazenar todos os arquivos do caminho path que são cvs, caso eu armazene algum arquivo extra na pasta dos datasets que não seja csv, não sera armazenado nessa variável.
 3. Criei um dicionário de dataframes, onde a chave é o nome do tabela e o valor é o próprio dataframe. Exemplo de acesso em um dos df abaixo.

In [4]:
# Visualizando os df
dataframes['olist_customers_dataset']

customer_id                customer_unique_id  \
0      06b8999e2fba1a1fbc88172c00ba8bc7  861eff4711a542e4b93843c6dd7febb0   
1      18955e83d337fd6b2def6b18a428ac77  290c77bc529b7ac935b93aa66c333dc3   
2      4e7b3e00288586ebd08712fdd0374a03  060e732b5b29e8181a18229c7b0b2b5e   
3      b2b6027bc5c5109e529d4dc6358b12c3  259dac757896d24d7702b9acbbff3f3c   
4      4f2d8ab171c80ec8364f7c12e35b23ad  345ecd01c38d18a9036ed96c73b8d066   
...                                 ...                               ...   
99436  17ddf5dd5d51696bb3d7c6291687be6f  1a29b476fee25c95fbafc67c5ac95cf8   
99437  e7b71a9017aa05c9a7fd292d714858e8  d52a67c98be1cf6a5c84435bd38d095d   
99438  5e28dfe12db7fb50a4b2f691faecea5e  e9f50caf99f032f0bf3c55141f019d99   
99439  56b18e2166679b8a959d72dd06da27f9  73c2643a0a458b49f58cea58833b192e   
99440  274fa6071e5e17fe303b9748641082c8  84732c5050c01db9b23e19ba39899398   

       customer_zip_code_prefix          customer_city customer_state  
0                         14409                 franca             SP  
1                          9790  sao bernardo do campo             SP  
2                          1151              sao paulo             SP  
3                          8775        mogi das cruzes             SP  
4                         13056               campinas             SP  
...                         ...                    ...            ...  
99436                      3937              sao paulo             SP  
99437                      6764        taboao da serra             SP  
99438                     60115              fortaleza             CE  
99439                     92120                 canoas             RS  
99440                      6703                  cotia             SP  

[99441 rows x 5 columns]

## Limpeza dos dados:

In [5]:
# Verificando nulos
for tabela, df in dataframes.items():
    print(f"Nulos na tabela: {tabela}")
    print(df.isnull().sum())
    print("---------------" * 10)


Nulos na tabela: olist_order_payments_dataset
order_id                0
payment_sequential      0
payment_type            0
payment_installments    0
payment_value           0
dtype: int64
------------------------------------------------------------------------------------------------------------------------------------------------------
Nulos na tabela: olist_order_items_dataset
order_id               0
order_item_id          0
product_id             0
seller_id              0
shipping_limit_date    0
price                  0
freight_value          0
dtype: int64
------------------------------------------------------------------------------------------------------------------------------------------------------
Nulos na tabela: olist_orders_dataset
order_id                            0
customer_id                         0
order_status                        0
order_purchase_timestamp            0
order_approved_at                 160
order_delivered_carrier_date     1783
order_delive

## Temos nulos em 3 tabelas:
1. olist_orders_dataset
2. olist_products_dataset
3. olist_order_reviews_dataset

Vou explorar cada tabela individualmente para identificar o que cada nulo quer dizer

# Olist_orders_dataset

In [6]:
dataframes['olist_orders_dataset']['order_approved_at']

0        2017-10-02 11:07:15
1        2018-07-26 03:24:27
2        2018-08-08 08:55:23
3        2017-11-18 19:45:59
4        2018-02-13 22:20:29
                ...         
99436    2017-03-09 09:54:05
99437    2018-02-06 13:10:37
99438    2017-08-27 15:04:16
99439    2018-01-08 21:36:21
99440    2018-03-09 11:20:28
Name: order_approved_at, Length: 99441, dtype: object

In [7]:
dataframes['olist_orders_dataset']['order_delivered_carrier_date']

0        2017-10-04 19:55:00
1        2018-07-26 14:31:00
2        2018-08-08 13:50:00
3        2017-11-22 13:39:59
4        2018-02-14 19:46:34
                ...         
99436    2017-03-10 11:18:03
99437    2018-02-07 23:22:42
99438    2017-08-28 20:52:26
99439    2018-01-12 15:35:03
99440    2018-03-09 22:11:59
Name: order_delivered_carrier_date, Length: 99441, dtype: object

In [8]:
dataframes['olist_orders_dataset']['order_delivered_customer_date']

0        2017-10-10 21:25:13
1        2018-08-07 15:27:45
2        2018-08-17 18:06:29
3        2017-12-02 00:28:42
4        2018-02-16 18:17:02
                ...         
99436    2017-03-17 15:08:01
99437    2018-02-28 17:37:56
99438    2017-09-21 11:24:17
99439    2018-01-25 23:32:54
99440    2018-03-16 13:08:30
Name: order_delivered_customer_date, Length: 99441, dtype: object

## Cada coluna armazena a data e hora respectiva ao seu nome
1. order_approved_at - Data e Hora que o pedido foi aprovado, se está nulos, temos pedidos que não foram aprovados ou que tiveram algum tipo de erro em sua aprovação
2. order_delivered_carrier_date: Data e Hora que o pedido foi enviado, se temos nulos pode-se dizer que o pedido não foi enviado ou que teve problemas de envio
3. order_delivered_customer_date: Data e hora que o pedido foi entregue, se não temos pode-se dizer que o pedido não chegou ao cliente.

Vale ressaltar que as colunas estao em formato object, transformarei as colunas em datetime em todas as tabelas necessárias enquanto faço o tratamento de nulos.

Na primeira coluna, temos apenas 160 dados nulos, como temos um dataset de 100000 dados, 160 dados acaba não sendo tão relevante para nossa análise, dessa forma irei excluir os nulos da tabela order_approved_at

Ja nas 2 tabelas seguintes, temos 1700 e 3000 dados. Ou seja, cerca de 1,7%-3% do dataset completo, tendo em vista que também podemos utilizar esses dados depois para verificar entregas e análises das mesmas farei duas colunas novas sendo elas
1. "enviados", com valores 1 para pedidos enviados, 0 para pedidos nao enviado(nulos).
2. "entregues": com valores 1 para os pedidos entregues e 0 para os valores nao entregues(nulos). 

E manterei os nulos nas colunas originais

In [9]:
# Excluir dados da tabela order_approved_at
dataframes['olist_orders_dataset'] = dataframes['olist_orders_dataset'].dropna(subset=['order_approved_at'])

dataframes['olist_orders_dataset']['order_approved_at'].isnull().sum()

np.int64(0)

In [10]:
# Object para datetime
dataframes['olist_orders_dataset']['order_approved_at'] = pd.to_datetime(dataframes['olist_orders_dataset']['order_approved_at'], errors='coerce')
dataframes['olist_orders_dataset']['order_delivered_carrier_date'] = pd.to_datetime(dataframes['olist_orders_dataset']['order_delivered_carrier_date'], errors='coerce')
dataframes['olist_orders_dataset']['order_delivered_customer_date'] = pd.to_datetime(dataframes['olist_orders_dataset']['order_delivered_customer_date'], errors='coerce')

In [11]:
# Criacao de novas colunas
dataframes['olist_orders_dataset']['enviados'] = dataframes['olist_orders_dataset']['order_delivered_carrier_date'].notna().astype(int)
dataframes['olist_orders_dataset']['entregues'] = dataframes['olist_orders_dataset']['order_delivered_customer_date'].notna().astype(int)

In [12]:
dataframes['olist_orders_dataset']['entregues'].value_counts()

entregues
1    96462
0     2819
Name: count, dtype: int64

In [13]:
dataframes['olist_orders_dataset']['enviados'].value_counts()

enviados
1    97644
0     1637
Name: count, dtype: int64

# olist_products_dataset

1. product_category_name         610
2. product_name_lenght           610
3. product_description_lenght    610
4. product_photos_qty            610
5. product_weight_g                2
6. product_length_cm               2
7. product_height_cm               2
8. product_width_cm                2

Temos valores muito pequenos para influenciar na análise, e por apresentarem o mesmo valor, provavelmente são referente há um mesmo tipo de produto produto, tendo isso em vista excluirei, todos os dados nulos dessa tabela:

In [14]:
df1 = dataframes['olist_products_dataset']
filtro_nulos = df1[df1['product_category_name'].isna()]


In [15]:
filtro_nulos

product_id product_category_name  \
105    a41e356c76fab66334f36de622ecbd3a                   NaN   
128    d8dee61c2034d6d075997acef1870e9b                   NaN   
145    56139431d72cd51f19eb9f7dae4d1617                   NaN   
154    46b48281eb6d663ced748f324108c733                   NaN   
197    5fb61f482620cb672f5e586bb132eae9                   NaN   
...                                 ...                   ...   
32515  b0a0c5dd78e644373b199380612c350a                   NaN   
32589  10dbe0fbaa2c505123c17fdc34a63c56                   NaN   
32616  bd2ada37b58ae94cc838b9c0569fecd8                   NaN   
32772  fa51e914046aab32764c41356b9d4ea4                   NaN   
32852  c4ceee876c82b8328e9c293fa0e1989b                   NaN   

       product_name_lenght  product_description_lenght  product_photos_qty  \
105                    NaN                         NaN                 NaN   
128                    NaN                         NaN                 NaN   
145                    NaN                         NaN                 NaN   
154                    NaN                         NaN                 NaN   
197                    NaN                         NaN                 NaN   
...                    ...                         ...                 ...   
32515                  NaN                         NaN                 NaN   
32589                  NaN                         NaN                 NaN   
32616                  NaN                         NaN                 NaN   
32772                  NaN                         NaN                 NaN   
32852                  NaN                         NaN                 NaN   

       product_weight_g  product_length_cm  product_height_cm  \
105               650.0               17.0               14.0   
128               300.0               16.0                7.0   
145               200.0               20.0               20.0   
154             18500.0               41.0               30.0   
197               300.0               35.0                7.0   
...                 ...                ...                ...   
32515            1800.0               30.0               20.0   
32589             800.0               30.0               10.0   
32616             200.0               21.0                8.0   
32772            1300.0               45.0               16.0   
32852             700.0               28.0                3.0   

       product_width_cm  
105                12.0  
128                20.0  
145                20.0  
154                41.0  
197                12.0  
...                 ...  
32515              70.0  
32589              23.0  
32616              16.0  
32772              45.0  
32852              43.0  

[610 rows x 9 columns]

In [16]:
dataframes['olist_products_dataset'] = dataframes['olist_products_dataset'].dropna(
    subset=['product_category_name', 'product_name_lenght', 'product_description_lenght', 'product_photos_qty'])


In [17]:
dataframes['olist_products_dataset'].isnull().sum()

product_id                    0
product_category_name         0
product_name_lenght           0
product_description_lenght    0
product_photos_qty            0
product_weight_g              1
product_length_cm             1
product_height_cm             1
product_width_cm              1
dtype: int64

In [18]:
dataframes['olist_products_dataset'].dropna(inplace=True)


/tmp/ipykernel_3954/1210711815.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dataframes['olist_products_dataset'].dropna(inplace=True)


In [19]:
dataframes['olist_products_dataset'].isnull().sum()

product_id                    0
product_category_name         0
product_name_lenght           0
product_description_lenght    0
product_photos_qty            0
product_weight_g              0
product_length_cm             0
product_height_cm             0
product_width_cm              0
dtype: int64

In [20]:
dataframes['olist_products_dataset']

product_id              product_category_name  \
0      1e9e8ef04dbcff4541ed26657ea517e5                         perfumaria   
1      3aa071139cb16b67ca9e5dea641aaa2f                              artes   
2      96bd76ec8810374ed1b65e291975717f                      esporte_lazer   
3      cef67bcfe19066a932b7673e239eb23d                              bebes   
4      9dc1a7de274444849c219cff195d0b71              utilidades_domesticas   
...                                 ...                                ...   
32946  a0b7d5a992ccda646f2d34e418fff5a0                   moveis_decoracao   
32947  bf4538d88321d0fd4412a93c974510e6  construcao_ferramentas_iluminacao   
32948  9a7c6041fa9592d9d9ef6cfe62a71f8c                    cama_mesa_banho   
32949  83808703fc0706a22e264b9d75f04a2e             informatica_acessorios   
32950  106392145fca363410d287a815be6de4                    cama_mesa_banho   

       product_name_lenght  product_description_lenght  product_photos_qty  \
0                     40.0                       287.0                 1.0   
1                     44.0                       276.0                 1.0   
2                     46.0                       250.0                 1.0   
3                     27.0                       261.0                 1.0   
4                     37.0                       402.0                 4.0   
...                    ...                         ...                 ...   
32946                 45.0                        67.0                 2.0   
32947                 41.0                       971.0                 1.0   
32948                 50.0                       799.0                 1.0   
32949                 60.0                       156.0                 2.0   
32950                 58.0                       309.0                 1.0   

       product_weight_g  product_length_cm  product_height_cm  \
0                 225.0               16.0               10.0   
1                1000.0               30.0               18.0   
2                 154.0               18.0                9.0   
3                 371.0               26.0                4.0   
4                 625.0               20.0               17.0   
...                 ...                ...                ...   
32946           12300.0               40.0               40.0   
32947            1700.0               16.0               19.0   
32948            1400.0               27.0                7.0   
32949             700.0               31.0               13.0   
32950            2083.0               12.0                2.0   

       product_width_cm  
0                  14.0  
1                  20.0  
2                  15.0  
3                  26.0  
4                  13.0  
...                 ...  
32946              40.0  
32947              16.0  
32948              27.0  
32949              20.0  
32950               7.0  

[32340 rows x 9 columns]

# olist_order_reviews_dataset

- Nulos na tabela: olist_order_reviews_dataset
- review_id                      0
- order_id                       0
- review_score                   0
- review_comment_title       87656
- review_comment_message     58247
- review_creation_date           0
- review_answer_timestamp        0
- dtype: int64

Como podemos observar no dataset abaixo, os nulos são referentes as reviews dos clientes, que são bem comuns de não serem preenchidas, dado esse cenário optarei por manter os nulos

In [21]:
dataframes['olist_order_reviews_dataset']

review_id                          order_id  \
0      7bc2406110b926393aa56f80a40eba40  73fc7af87114b39712e6da79b0a377eb   
1      80e641a11e56f04c1ad469d5645fdfde  a548910a1c6147796b98fdf73dbeba33   
2      228ce5500dc1d8e020d8d1322874b6f0  f9e4b658b201a9f2ecdecbb34bed034b   
3      e64fb393e7b32834bb789ff8bb30750e  658677c97b385a9be170737859d3511b   
4      f7c4243c7fe1938f181bec41a392bdeb  8e6bfb81e283fa7e4f11123a3fb894f1   
...                                 ...                               ...   
99219  574ed12dd733e5fa530cfd4bbf39d7c9  2a8c23fee101d4d5662fa670396eb8da   
99220  f3897127253a9592a73be9bdfdf4ed7a  22ec9f0669f784db00fa86d035cf8602   
99221  b3de70c89b1510c4cd3d0649fd302472  55d4004744368f5571d1f590031933e4   
99222  1adeb9d84d72fe4e337617733eb85149  7725825d039fc1f0ceb7635e3f7d9206   
99223  efe49f1d6f951dd88b51e6ccd4cc548f  90531360ecb1eec2a1fbb265a0db0508   

       review_score review_comment_title  \
0                 4                  NaN   
1                 5                  NaN   
2                 5                  NaN   
3                 5                  NaN   
4                 5                  NaN   
...             ...                  ...   
99219             5                  NaN   
99220             5                  NaN   
99221             5                  NaN   
99222             4                  NaN   
99223             1                  NaN   

                                  review_comment_message review_creation_date  \
0                                                    NaN  2018-01-18 00:00:00   
1                                                    NaN  2018-03-10 00:00:00   
2                                                    NaN  2018-02-17 00:00:00   
3                  Recebi bem antes do prazo estipulado.  2017-04-21 00:00:00   
4      Parabéns lojas lannister adorei comprar pela I...  2018-03-01 00:00:00   
...                                                  ...                  ...   
99219                                                NaN  2018-07-07 00:00:00   
99220                                                NaN  2017-12-09 00:00:00   
99221  Excelente mochila, entrega super rápida. Super...  2018-03-22 00:00:00   
99222                                                NaN  2018-07-01 00:00:00   
99223  meu produto chegou e ja tenho que devolver, po...  2017-07-03 00:00:00   

      review_answer_timestamp  
0         2018-01-18 21:46:59  
1         2018-03-11 03:05:13  
2         2018-02-18 14:36:24  
3         2017-04-21 22:02:06  
4         2018-03-02 10:26:53  
...                       ...  
99219     2018-07-14 17:18:30  
99220     2017-12-11 20:06:42  
99221     2018-03-23 09:10:43  
99222     2018-07-02 12:59:13  
99223     2017-07-03 21:01:49  

[99224 rows x 7 columns]

# Vamos observar abaixo a quantidade de duplicatas nas tabelas do dicionario de df

In [23]:
# Verificando duplicatas
for tabela, df in dataframes.items():
    print(f"Duplicatas na tabela: {tabela}")
    print(df.duplicated().sum())
    print("---------------" * 10)


Duplicatas na tabela: olist_order_payments_dataset
0
------------------------------------------------------------------------------------------------------------------------------------------------------
Duplicatas na tabela: olist_order_items_dataset
0
------------------------------------------------------------------------------------------------------------------------------------------------------
Duplicatas na tabela: olist_orders_dataset
0
------------------------------------------------------------------------------------------------------------------------------------------------------
Duplicatas na tabela: product_category_name_translation
0
------------------------------------------------------------------------------------------------------------------------------------------------------
Duplicatas na tabela: olist_products_dataset
0
-----------------------------------------------------------------------------------------------------------------------------------------------

In [24]:
dataframes['olist_geolocation_dataset']

geolocation_zip_code_prefix  geolocation_lat  geolocation_lng  \
0                               1037       -23.545621       -46.639292   
1                               1046       -23.546081       -46.644820   
2                               1046       -23.546129       -46.642951   
3                               1041       -23.544392       -46.639499   
4                               1035       -23.541578       -46.641607   
...                              ...              ...              ...   
1000158                        99950       -28.068639       -52.010705   
1000159                        99900       -27.877125       -52.224882   
1000160                        99950       -28.071855       -52.014716   
1000161                        99980       -28.388932       -51.846871   
1000162                        99950       -28.070104       -52.018658   

        geolocation_city geolocation_state  
0              sao paulo                SP  
1              sao paulo                SP  
2              sao paulo                SP  
3              sao paulo                SP  
4              sao paulo                SP  
...                  ...               ...  
1000158         tapejara                RS  
1000159   getulio vargas                RS  
1000160         tapejara                RS  
1000161  david canabarro                RS  
1000162         tapejara                RS  

[1000163 rows x 5 columns]

In [25]:
dataframes['olist_geolocation_dataset'][dataframes['olist_geolocation_dataset'].duplicated(keep=False)]

geolocation_zip_code_prefix  geolocation_lat  geolocation_lng  \
0                               1037       -23.545621       -46.639292   
1                               1046       -23.546081       -46.644820   
2                               1046       -23.546129       -46.642951   
6                               1047       -23.546273       -46.641225   
7                               1013       -23.546923       -46.634264   
...                              ...              ...              ...   
1000153                        99970       -28.343273       -51.873734   
1000154                        99950       -28.070493       -52.011342   
1000159                        99900       -27.877125       -52.224882   
1000160                        99950       -28.071855       -52.014716   
1000162                        99950       -28.070104       -52.018658   

        geolocation_city geolocation_state  
0              sao paulo                SP  
1              sao paulo                SP  
2              sao paulo                SP  
6              sao paulo                SP  
7              sao paulo                SP  
...                  ...               ...  
1000153          ciriaco                RS  
1000154         tapejara                RS  
1000159   getulio vargas                RS  
1000160         tapejara                RS  
1000162         tapejara                RS  

[390005 rows x 5 columns]

In [26]:
# Ordenarei as duplicatas para ficarem juntas
geoloc = dataframes['olist_geolocation_dataset'].sort_values(by=dataframes['olist_geolocation_dataset'].columns.tolist())

# Mostrar apenas duplicatas ordenadas
geoloc[geoloc.duplicated(keep=False)]


geolocation_zip_code_prefix  geolocation_lat  geolocation_lng  \
519                             1001       -23.551337       -46.634027   
583                             1001       -23.551337       -46.634027   
818                             1001       -23.551337       -46.634027   
206                             1001       -23.550498       -46.634338   
429                             1001       -23.550498       -46.634338   
...                              ...              ...              ...   
999899                         99980       -28.387432       -51.847727   
999958                         99980       -28.387059       -51.848964   
1000129                        99980       -28.387059       -51.848964   
999775                         99980       -28.386689       -51.847091   
1000133                        99980       -28.386689       -51.847091   

        geolocation_city geolocation_state  
519            sao paulo                SP  
583            sao paulo                SP  
818            sao paulo                SP  
206            sao paulo                SP  
429            sao paulo                SP  
...                  ...               ...  
999899   david canabarro                RS  
999958   david canabarro                RS  
1000129  david canabarro                RS  
999775   david canabarro                RS  
1000133  david canabarro                RS  

[390005 rows x 5 columns]

## Temos exatos dados nulos na tabela.
É normal termos o mesmo cep no nosso dataset, afinal mais de uma pessoa pode morar no mesmo endereço em casas diferentes, que resultariam diretamente no mesmo cep, que na nossa tabela esta definida como zip code, porém como os dados de geolocalização são identicos temos que estão indexados na mesma casa, e por isso irei remover tais dados para limpar os dados idênticos.

In [27]:
# Excluindo duplicatas
dataframes['olist_geolocation_dataset'] = dataframes['olist_geolocation_dataset'].drop_duplicates()

In [31]:
# agora temos os dados limpos de duplicatas.
dataframes['olist_geolocation_dataset'].duplicated().sum()

np.int64(0)

### Farei a normalização das colunas
Para fazer isso irei utilizar o modulo info no dataset para conseguir visualizar os tipos de colunas em que podem ou devem ser alteradas



In [35]:
for tabela, df in dataframes.items():
    print(f'Nome da tabela: {tabela}')
    print(df.info())
    print('-----' * 10)

Nome da tabela: olist_order_payments_dataset
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 103886 entries, 0 to 103885
Data columns (total 5 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   order_id              103886 non-null  object 
 1   payment_sequential    103886 non-null  int64  
 2   payment_type          103886 non-null  object 
 3   payment_installments  103886 non-null  int64  
 4   payment_value         103886 non-null  float64
dtypes: float64(1), int64(2), object(2)
memory usage: 4.0+ MB
None
--------------------------------------------------
Nome da tabela: olist_order_items_dataset
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 112650 entries, 0 to 112649
Data columns (total 7 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   order_id             112650 non-null  object 
 1   order_item_id        112650 non-null  int64  
 2   pr

In [37]:
# Os tipos dos dados nas tabelas estão normalizados
dataframes['olist_order_payments_dataset'].head()

order_id  payment_sequential payment_type  \
0  b81ef226f3fe1789b1e8b2acac839d17                   1  credit_card   
1  a9810da82917af2d9aefd1278f1dcfa0                   1  credit_card   
2  25e8ea4e93396b6fa0d3dd708e76c1bd                   1  credit_card   
3  ba78997921bbcdc1373bb41e913ab953                   1  credit_card   
4  42fdf880ba16b47b59251dd489d4441a                   1  credit_card   

   payment_installments  payment_value  
0                     8          99.33  
1                     1          24.39  
2                     1          65.71  
3                     8         107.78  
4                     2         128.45

In [38]:
dataframes['olist_order_items_dataset'].head()

order_id  order_item_id  \
0  00010242fe8c5a6d1ba2dd792cb16214              1   
1  00018f77f2f0320c557190d7a144bdd3              1   
2  000229ec398224ef6ca0657da4fc703e              1   
3  00024acbcdf0a6daa1e931b038114c75              1   
4  00042b26cf59d7ce69dfabb4e55b4fd9              1   

                         product_id                         seller_id  \
0  4244733e06e7ecb4970a6e2683c13e61  48436dade18ac8b2bce089ec2a041202   
1  e5f2d52b802189ee658865ca93d83a8f  dd7ddc04e1b6c2c614352b383efe2d36   
2  c777355d18b72b67abbeef9df44fd0fd  5b51032eddd242adc84c38acab88f23d   
3  7634da152a4610f1595efa32f14722fc  9d7a1d34a5052409006425275ba1c2b4   
4  ac6c3623068f30de03045865e4e10089  df560393f3a51e74553ab94004ba5c87   

   shipping_limit_date   price  freight_value  
0  2017-09-19 09:45:35   58.90          13.29  
1  2017-05-03 11:05:13  239.90          19.93  
2  2018-01-18 14:48:30  199.00          17.87  
3  2018-08-15 10:10:18   12.99          12.79  
4  2017-02-13 13:57:51  199.90          18.14

In [39]:
# temos a coluna shipping_limit_date como object, transformarei a mesma em datetime
dataframes['olist_order_items_dataset']['shipping_limit_date'] = pd.to_datetime(dataframes['olist_order_items_dataset']['shipping_limit_date'], errors='coerce')


In [40]:
dataframes['olist_order_items_dataset'].info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 112650 entries, 0 to 112649
Data columns (total 7 columns):
 #   Column               Non-Null Count   Dtype         
---  ------               --------------   -----         
 0   order_id             112650 non-null  object        
 1   order_item_id        112650 non-null  int64         
 2   product_id           112650 non-null  object        
 3   seller_id            112650 non-null  object        
 4   shipping_limit_date  112650 non-null  datetime64[ns]
 5   price                112650 non-null  float64       
 6   freight_value        112650 non-null  float64       
dtypes: datetime64[ns](1), float64(2), int64(1), object(3)
memory usage: 6.0+ MB
